sfy-processing/sfy/cli/collection.py

```diff
@@ -12,27 +17,130 @@ def collection():
 @collection.command()
 @click.argument('config', type=click.File())
 def archive(config):
+    """Create CF-compatible trajectory file based on yml configuation file
+
+    Presently only works for Floatensteins
+    """
+
     logger.info(f'Reading configuration file: {config.name}')
 
-    hub = Hub.from_env()
-    print(hub.buoys())
+    with open(config.name, 'r') as f:
+        config = yaml.safe_load(f)
 
-    b = hub.buoy('2022_CIRFA_JR_drifter_1')
-    print(b)
+    hub = Hub.from_env()
+    # Select only the buoys that are listed in config file
+    buoys = [b for b in hub.buoys() if b.dev in config['drifters']]
 
-    p = b.packages()
-    print(p)
+    dicts = {}  # to hold all messages
 
-    j = b.json_package(p[0][0])
-    print(j)
+    for b in buoys:
+        dicts[b] = {}
+        packages = b.fetch_packages_range(start=datetime(1, 1, 1), end=datetime.now())
+        for p in packages:
+            j = json.loads(p[2])
+            for m in j['body']['messages']:
+                if 'latitude' in m:
+                    dicts[b][datetime.fromtimestamp(m['datetime_fix'])] = \
+                        {'lon': m['longitude'], 'lat': m['latitude']}
 
-    # config = yaml.load(config)
-    # print(config)
+    ds = trajectory_dict_to_dataset(dicts, config['attributes'])
+    ds.to_netcdf(f"{config['name']}.nc")
 
 @collection.command()
 @click.argument('config', type=click.File('w'))
-def template(config):
+@click.option('-f', '--filter',
+              default=None,
+              help='Filter on drifter names (case insensitive)',
+              type=str,
+              multiple=True)
+
+def template(config, filter):
+    """Create template yml file which can be manually edited before creating netCDF"""
+
     logger.info(f'Writing configuration file: {config.name}')
 
-    # config = yaml.load(config)
-    # print(config)
+    # Template yaml. Can add more standard attributes here, to be filled in config file by user
+    t = {
+        'name': f'{config.name.split(".")[0]}',
+        'attributes': {
+            'title': '',
+            'summary': '',
+            'history': '',
+            'creator_name': '',
+            'creator_email': '',
+            'creator_url': '',
+            'references': '',
+            }
+        }
+
+    # Fetch list of drifters
+    hub = Hub.from_env()
+    b0 = hub.buoys()[0]
+    t['drifters'] = [b.dev for b in hub.buoys()]
+    if filter != ():
+        for filterstring in filter:
+            t['drifters'] = [b for b in t['drifters'] if filterstring.lower() in b.lower()]
+
+    # Unfortunately order is not preserved, as OrderedDict is not supported:
+    # https://stackoverflow.com/questions/5121931/in-python-how-can-you-load-yaml-mappings-as-ordereddicts)
+    yaml.dump(t, open(config.name, 'w'))
+
+# This method might be moved to trajectory_analysis package?
```

Comment:
Do you agree, @gauteh ?

Reply:
Presently trajectory_analysis package is part of OpenDrift, but I guess the plan is to make it stand-alone(?)
Is it ok for Sfy to depend on OpenDrift (for this functionality) for now?

Reply:
Yes. That's completely fine. Maybe we should start putting it in a new package already, since the t-a lives in a different branch now?

Reply:
Yes, name is important. Might not be possible to change package name after creation(?) so should think first. I start thinking now.

Reply:
After a little thinking, I suggest to work more on the trajectory_analysis "package" in OpenDrift first, to make if clearer what it will contain, and thus what will be a good name.

Reply:
Quickly and we'll thought.

Reply:
It is possible to change it, before we publish it to pip and conda.